store/sqlstore/remote_cluster_store.go

```diff
@@ -54,7 +79,22 @@ func (s sqlRemoteClusterStore) Update(remoteCluster *model.RemoteCluster) (*mode
 		return nil, err
 	}
 
-	if _, err := s.GetMaster().Update(remoteCluster); err != nil {
+	query := `UPDATE RemoteClusters
+			  SET Name = :Name,
+			  Token = :Token,
+			  RemoteToken = :RemoteToken,
+			  CreatorId = :CreatorId,
+			  DisplayName = :DisplayName,
+			  SiteURL = :SiteURL,
+			  Topics = :Topics
+			  WHERE	RemoteId = :RemoteId`
+
+	err := s.ValidSiteURL(*remoteCluster)
```

Comment:
There's no need for this. The update should always happen by the primary key, which in this case is a compound key. So let's move the `Name` as well in the `WHERE` clause.

Reply:
By removing the validation, the tests still fails because the Pkey is correct. In the `remote_cluster_test`, We test by having a model with the same `RemoteTeamId` and `SiteUrl` which does not go against the Primary Key (Name, SiteUrl)

Reply:
That is because we were not updating all the columns :)

If we fix the query to

```diff
        query := `UPDATE RemoteClusters
-                         SET Token = :Token,
-                         RemoteToken = :RemoteToken,
-                         CreatorId = :CreatorId,
+                         SET RemoteTeamId = :RemoteTeamId,
                          DisplayName = :DisplayName,
                          SiteURL = :SiteURL,
+                         CreateAt = :CreateAt,
+                         LastPingAt = :LastPingAt,
+                         Token = :Token,
+                         RemoteToken = :RemoteToken,
+                         CreatorId = :CreatorId,
                          Topics = :Topics
                          WHERE RemoteId = :RemoteId AND Name = :Name`

```

the tests pass